packages/server/src/fhir/operations/deploy.ts

```diff
@@ -12,47 +13,62 @@ export async function deployHandler(req: FhirRequest): Promise<FhirResponse> {
   const ctx = getAuthenticatedContext();
   const { id } = req.params;
 
-  // Validate that the request body has a code property
-  const code = req.body.code as string | undefined;
-  if (!code) {
-    return [badRequest('Missing code')];
-  }
-
   // First read the bot as the user to verify access
   await ctx.repo.readResource<Bot>('Bot', id);
 
   // Then read the bot as system user to load extended metadata
   const systemRepo = getSystemRepo();
   const bot = await systemRepo.readResource<Bot>('Bot', id);
 
+  // Validate that the request body has a code property
+  // Or that the Bot already has executable code attached
+  const code = req.body.code as string | undefined;
```

Comment:
Moved this block because we need to get the `bot` before doing this check now, which requires the `readResource` to come first